controllers/provisioners/eks/helpers.go

```diff
@@ -156,9 +162,20 @@ func (ctx *EksInstanceGroupContext) GetLabelList() []string {
 	}
 	sort.Strings(labelList)
 
-	// add role label
-	for _, label := range RoleLabelsFmt {
-		labelList = append(labelList, fmt.Sprintf(label, instanceGroup.GetName()))
+	// add role label according to the cluster's k8s version
+	addOldStyleLabel := false
+	clusterVersion, err := ctx.DiscoveredState.GetClusterVersion()
+	if err == nil {
+		c, err := semver.NewConstraint("< 1.16-0")
```

Comment:
what does `-0` mean in this case?
I think we can rely on the EKS version being only major.minor e.g. `1.16` or `1.15`

Reply:
Another possible idiom which can work here is:
```
if c, err := semver.NewConstraint("< 1.16-0"); err != nil {
  labelList = append(labelList, fmt.Sprintf(RoleNewLabelFmt, instanceGroup.GetName()))
}
```

Reply:
we don't really need to handle the case of an error here, just don't append anything..
we don't really expect an error to occur here, it would mean there is no cluster version populated?

Reply:
Sure, that's fine to keep it in that case